frontend/server/src/main/java/org/pytorch/serve/wlm/ModelManager.java

```diff
@@ -205,87 +204,139 @@ private ModelArchive createModelArchive(
         return archive;
     }
 
+    private void setupModelVenv(Model model)
+            throws IOException, InterruptedException, ModelException {
+        String venvPath = EnvironmentUtils.getPythonVenvPath(model);
+        List<String> commandParts = new ArrayList<>();
+        commandParts.add(EnvironmentUtils.getPythonRunTime(model));
+        commandParts.add("-m");
+        commandParts.add("venv");
+        commandParts.add("--clear");
+        commandParts.add("--system-site-packages");
```

Comment:
why system site?

Reply:
This is to make sure that the `venv` can see packages that are already installed in the base python environment and not have to install them again. If a newer version of an existing package is required or a non existing package is required, they will get installed to the venv `site-packages` and will take precedence over `system-site-packages`. This does not affect the base python environment. I've added more details with an example in the PR description: https://github.com/pytorch/serve/pull/2910#issue-2103056259